src/Components/Components/src/ComponentBase.cs

```diff
@@ -148,15 +148,15 @@ protected virtual Task OnAfterRenderAsync()
         /// </summary>
         /// <param name="workItem">The work item to execute.</param>
         protected Task InvokeAsync(Action workItem)
-            => _renderHandle.InvokeAsync(workItem);
+            => _renderHandle.Dispatcher.InvokeAsync(workItem);
```

Comment:
Looking for feedback on whether to keep these or go all the way and expose the dispatcher directly here (and remove these). 

The way things are right now, you don't need to know about the `Dispatcher` as a component author, but also can't easily get at it (unless you bypass `ComponentBase`). We could also choose to expose the dispatcher and keep these convenience APIs.

Reply:
I prefer what you've done here already. For most people building on `ComponentBase`, it's good to hide the `Dispatcher` so people don't trip up with mistakes like passing references to dispatcher to nonsensical places like domain model types. I'm glad the only way to access it is via `protected` methods on `ComponentBase`, so you really have to go out of your way if you want to expose public access to it or capture delegates that can invoke it.

The underlying `IComponent` level is intended to be more bare-metal and unopinionated about how you interact with the framework so it does feel right that this is the level you have to be at in order to mess with `Dispatcher` directly.